### ISSUE TITLE
Generate appropriate description for code flow PRs

### ISSUE THREAD

**Issue by premun** [dotnet/arcade-services#3364]
## Context

We paid no attention to the content (title, description) of the PR so far and put a very basic info that matches what we put on the usual PRs.
We should fill the description with relevant info to the code flow update (from-to SHAs, links to diffs..).

## Goals 
Once we start creating PRs, we need to generate the right content for their description. We need to update the title/description with each update too.

- We need to link the diff of sources that are being synchronized from the original repo.
- The build number should be clickable and lead to the Azure DevOps build that the issues links.
- ~We need to also put the list of dependencies we're synchronizing.~
- We should put a link to some documentation on what the PRs are, how to resolve conflicts etc.
- ~Batchable PRs must contain all of the updates (there's a Scenario test that creates a batchable subscription to test this on).~

**Comment by premun** [comment]
There was an ask from Viktor to make sure the commit SHA is a clickable link

**Comment by adamzip** [comment]
Revised goals for this issue:
- Add a commit link and build link to **non-codeflow** PRs and codeflow PRs. Link should work for both GH and Azdo
- Add a clickable link in the header that shows the codeflow diff since last synced commit
- Update codeflow PRs when updates occur (new commit, new batchable build trigger, etc
- Add a link to codeflow documentation

Out of scope (these goals will be completed in a different issue):
- Add a list of dependencies (blocked by WIP)
